### ISSUE TITLE
During the test, after changing the box position, the generated results are not good?

### ISSUE THREAD

**Issue by hitlxm** [HuiZhang0812/CreatiDesign#10]
During the test, only changing the box position, the generated results are not good?

The following are the original results. The input data is：

imgs_id:  ['20250222_104850']
objects_boxes:  tensor([[[0.7000, 0.0100, 0.9100, 0.3300],
         [0.0800, 0.1900, 0.6400, 0.3600],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000]]])
objects_caption:  [['A circular wreath composed of dried, spiky plant material on a neutral background.', 'text:GIFT THE MAGIC \nOF CHRISTMAS!']]

the result:

![Image](https://github.com/user-attachments/assets/96fcc6fd-2cd0-4ec5-9843-f0fb93b6e532)

Only adjust the position of the second box，The input data is：

imgs_id:  ['20250222_104850']
objects_boxes:  tensor([[[0.7000, 0.0100, 0.9100, 0.3300],
         [0.3000, 0.1900, 0.9000, 0.3600],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000],
         [0.0000, 0.0000, 0.0000, 0.0000]]])
objects_caption:  [['A circular wreath composed of dried, spiky plant material on a neutral background.', 'text:GIFT THE MAGIC \nOF CHRISTMAS!']]

the result:

![Image](https://github.com/user-attachments/assets/143b3e3c-bb7f-49e7-9dd2-8d7010b5acc3)

Finally, it seems that the text result will not be generated according to the box position. Is there something wrong with the settings?

**Comment by HuiZhang0812** [comment]
> During the test, only changing the box position, the generated results are not good?
> 
> The following are the original results. The input data is：
> 
> imgs_id: ['20250222_104850'] objects_boxes: tensor([[[0.7000, 0.0100, 0.9100, 0.3300], [0.0800, 0.1900, 0.6400, 0.3600], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000]]]) objects_caption: [['A circular wreath composed of dried, spiky plant material on a neutral background.', 'text:GIFT THE MAGIC \nOF CHRISTMAS!']]
> 
> the result:
> 
> ![Image](https://github.com/user-attachments/assets/96fcc6fd-2cd0-4ec5-9843-f0fb93b6e532)
> 
> Only adjust the position of the second box，The input data is：
> 
> imgs_id: ['20250222_104850'] objects_boxes: tensor([[[0.7000, 0.0100, 0.9100, 0.3300], [0.3000, 0.1900, 0.9000, 0.3600], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000], [0.0000, 0.0000, 0.0000, 0.0000]]]) objects_caption: [['A circular wreath composed of dried, spiky plant material on a neutral background.', 'text:GIFT THE MAGIC \nOF CHRISTMAS!']]
> 
> the result:
> 
> ![Image](https://github.com/user-attachments/assets/143b3e3c-bb7f-49e7-9dd2-8d7010b5acc3)
> 
> Finally, it seems that the text result will not be generated according to the box position. Is there something wrong with the settings?

**Comment by HuiZhang0812** [comment]
In the DesignDataset of dataloader.creatidesign_dataset_benchmark, we construct objects_masks_maps_padded based on the bounding box, which is used for the multimodal attention mask. Please confirm whether the box position you modified also works in the DesignDataset.

**Comment by hitlxm** [comment]
Thank you for your reply. It was my mistake. I can get the correct result after modifying the corresponding mask map.


I also implemented my own training code using 8 A100s and a batch size of 4. After 15,000 training steps, I got the following results: the text rendering is correct, but the foreground is not preserved.

![Image](https://github.com/user-attachments/assets/8fe6fe9a-c253-49bc-a2cb-f1c20afb3e00)

![Image](https://github.com/user-attachments/assets/3686cafd-ed62-49f7-b9ac-80f45b73493e)

![Image](https://github.com/user-attachments/assets/fe58e094-6514-4b24-bbc3-902b074694d9)

![Image](https://github.com/user-attachments/assets/ea81dc05-5c27-43c0-be34-b1d27ad7f990)

Based on the implementation details in your paper, "training for 100,000 steps with a batch size of 8 on 8 H20-96G GPUs," you trained for 8 x 8 x100,000 / 400,000 = 16 epochs?

So, is it because I didn't train enough steps above that I didn't get the correct results?
